lib/queue.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+var config = require('./config'),
+  q = require('q'),
+  _ = require('lodash');
+
+// meta data item model schema
```

Comment:
@nialldonnellyfh do you mean proposed model? This was just an example., however I have considered adding validations on required fields.